crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3416,7 +3461,7 @@ impl<'db> TypeInferenceBuilder<'db> {
             ExprContext::Store => {
                 let value_ty = self.infer_expression(value);
 
-                if let Type::Instance(instance) = value_ty {
+                let symbol = if let Type::Instance(instance) = value_ty {
```

Comment:
note for future: I suspect that we should allow member access to signal error conditions, rather than doing this special-casing based on the specific type out here in type inference